librz/arch/no_rtti.c

```diff
@@ -0,0 +1,170 @@
+			method.addr = vmethod->addr;
+			method.name = rz_str_dup(exist_meth.name);
+			method.real_name = rz_str_dup(exist_meth.real_name);
+			method.vtable_offset = vmethod->vtable_offset;
+			method.method_type = RZ_ANALYSIS_CLASS_METHOD_VIRTUAL;
+			rz_analysis_class_method_fini(&exist_meth);
+		} else {
+			rz_warn_if_reached();
+		}
+	}
+	return method;
+}
+
+/**
+ * \param analysis Rizin analysis object
+ * \param vtable_info Information about current virtual table
+ * \param addr_vect Vector of addresses to calculate xrefs to. For first call, this should have a single address pointing to vtable saadr
+ * \param method_class_map Map between methods and their class
+ * \param depth Recursion depth, for first call, it should be 0
+ * \brief Recursively calculates xrefs to virtual tables. Stops analysis when the first class method (in a branch) is discovered which uses the vtable.
+ */
+static void xrefs_to_vtables_recursive(RzAnalysis *analysis, RVTableInfo *vtable_info, RzVector /*<ut64>*/ *addr_vect, HtSS *method_class_map, ut16 depth) {
+	// Limiting the depth because class virtual table analysis is generally done in first few depths.
+	// This prevents redundant recursive analysis of vtables containing debug functions / non-virtual method vtables
+	if (rz_vector_len(addr_vect) == 0 || depth == 5) {
+		return;
+	}
+	RzVector *new_addr_vect = rz_vector_new(sizeof(ut64), NULL, NULL);
+	ut64 *it;
+
+	bool found = false;
+	rz_vector_foreach (addr_vect, it) {
+		ut64 addr = *it;
+		RzList *xref_list = rz_analysis_xrefs_get_to(analysis, addr);
+		if (!xref_list) {
+			continue;
+		}
+		RzListIter *itt;
+		RzAnalysisXRef *xref;
+		rz_list_foreach (xref_list, itt, xref) {
+			RzAnalysisFunction *func = rz_analysis_get_fcn_in(analysis, xref->from, 0);
+			if (!func) {
+				continue;
+			}
+			rz_vector_push(new_addr_vect, &func->addr);
+			const char *class_name = class_name_from_method(method_class_map, func);
+			if (!class_name) {
+				continue;
+			}
+			found = true;
+			RzAnalysisVTable vtable = { 0 };
+			vtable.addr = vtable_info->saddr;
+			RVTableMethodInfo *vmethod;
+			rz_vector_foreach (&vtable_info->methods, vmethod) {
+				RzAnalysisMethod method = get_class_method(analysis, vmethod, class_name);
+				rz_analysis_class_method_set(analysis, class_name, &method);
+				rz_analysis_class_method_fini(&method);
+			}
+			rz_analysis_class_vtable_set(analysis, class_name, &vtable);
+			rz_analysis_class_vtable_fini(&vtable);
+		}
+		rz_list_free(xref_list);
+	}
+	if (!found) {
+		xrefs_to_vtables_recursive(analysis, vtable_info, new_addr_vect, method_class_map, depth + 1);
+	}
+	rz_vector_free(new_addr_vect);
+}
+
+/**
+ * \param context Vtble context object
+ * \param vtables List of virtual tables
+ * \brief Runs non-rtti required analysis on binaries
+ */
+RZ_API void rz_analysis_no_rtti_analysis(RZ_NONNULL RVTableContext *context, RZ_NONNULL RzList /*<RVTableInfo *>*/ *vtables) {
```

Comment:
Sorry, last request. Please write more expressive doxygen here.

Also, first the `\vrief` then the `\param`